sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_customer_statsbeat.py

```diff
@@ -0,0 +1,538 @@
+                    "count": count
+                }
+                
+                if telemetry_type:
+                    new_entry["telemetry_type"] = telemetry_type
+                
+                if ((retry_code == RetryCode.CLIENT_EXCEPTION or 
+                     retry_code == "CLIENT_EXCEPTION" or
+                     retry_code == RetryCode.CLIENT_TIMEOUT) and exception_message):
+                    new_entry["exception_message"] = exception_message
+                
+                # Add retry.reason if provided
+                if retry_reason:
+                    new_entry["retry_reason"] = retry_reason
+                
+                counter.total_item_retry_count.append(new_entry)
+        except (TypeError, KeyError, AttributeError) as exc:
+            # Log specific exceptions that might occur during dictionary operations
+            logger.warning("Failed to count retry items due to a data error: %s", str(exc))
+        except Exception as exc:  # pylint: disable=broad-except
+            # Still catch all other exceptions to prevent affecting the application
+            # This is acceptable in statsbeat since it should never crash the host app
+            logger.warning("Failed to count retry items: %s", str(exc))
+    
+    def shutdown(self) -> None:
+        """Shutdown the customer statsbeat metrics."""
+        if not getattr(self, "_is_enabled", False):
+            return
+        
+        try:
+            # Shutdown the meter provider
+            if hasattr(self, '_customer_statsbeat_meter_provider'):
+                self._customer_statsbeat_meter_provider.shutdown()
+        except AttributeError as exc:
+            # Specific error when meter provider doesn't have shutdown
+            logger.warning("Error during statsbeat shutdown - attribute error: %s", str(exc))
+        except Exception as exc:  # pylint: disable=broad-except
+            # Still catch all other exceptions to prevent affecting the application            
+            # This is acceptable in statsbeat since it should never crash the host app
+            logger.warning("Error during statsbeat shutdown: %s", str(exc))
+    
+    def _item_success_callback(self, options: CallbackOptions) -> List[Observation]:
+        """Callback for reporting successful items.
+        
+        Args:
+            options: Callback options
+        
+        Returns:
+            List of observations
+        """
+        if not getattr(self, "_is_enabled", False):
+            return []
+
+        try:
+            observations = []
+            counter = getattr(self, "_customer_statsbeat_counter", None)
+            
+            # Check if counter exists and has the required attribute
+            if counter is None or not hasattr(counter, "total_item_success_count"):
+                return []
+            
+            # Create observations for each unique entry
+            for entry in counter.total_item_success_count:
```

Comment:
This is too inefficient when using a `List`. Use a `Dict[telemetry_type, count]` instead.